metron-interface/metron-rest/src/test/java/org/apache/metron/rest/controller/AlertsProfileControllerIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json("[" + user1ProfileJson + "]"));
+
+    this.mockMvc.perform(post(url).with(httpBasic(user2, password)).with(csrf())
+        .contentType(MediaType.parseMediaType("application/json;charset=UTF-8"))
+        .content(user2ProfileJson))
+        .andExpect(status().isCreated())
+        .andExpect(content().json(user2ProfileJson));
+
+    this.mockMvc.perform(get(url).with(httpBasic(user1, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json(user1ProfileJson));
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json(user2ProfileJson));
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json("[" + user1ProfileJson + "," + user2ProfileJson + "]"));
+
+    this.mockMvc.perform(delete(url + "/user1").with(httpBasic(admin, password)))
+        .andExpect(status().isOk());
+
+    this.mockMvc.perform(delete(url + "/user1").with(httpBasic(admin, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user1, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json(user2ProfileJson));
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json("[" + user2ProfileJson + "]"));
+
+    this.mockMvc.perform(delete(url + "/user2").with(httpBasic(admin, password)))
+        .andExpect(status().isOk());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user1, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
```

Comment:
This is not just consistent with the other integration tests in metron-rest, this is consistent with integration tests throughout Metron.  If you look at our integration tests across all modules, the majority of them only have 1 test method.  Integration tests are different than unit tests because they generally are simulating a use case, state changes between the individual tests, and tests depend on state.  Unit tests are different because each test usually sets up it's own state with mocks and test objects.

There are probably some other integration tests that we could break into smaller tests (security tests can be separate because they don't change state) but most of the time we can't.  I'm not sure I could move any of these tests because they all depend on previous endpoints being executed.  For example if I were to move the list users test to a different test it would no longer pass because it depends on the creation/deletion of users in other tests.

The only benefit I can see to doing this is to provide clarity on what is being tested.  If I added comments before each test explaining their purpose would that achieve the same result?